### ISSUE TITLE
Fail to build canvas on Mac Monterey

### ISSUE THREAD

**Issue by billcxx** [Automattic/node-canvas#1930]
## Issue or Feature
- [x] If this is an issue with installation, I have read the [troubleshooting guide](https://github.com/Automattic/node-canvas/issues/1511).

## Problem
I'm building an electronjs app that uses vtkjs. My app used to work fine, but after I upgrade to Mac Monterey, update Xcode, and `brew upgrade`, I can no longer build it. I can still build the exact same code on linux by the way. Here are some error print out:

```
$ electron-builder
  • electron-builder  version=22.13.1 os=21.1.0
  • loaded configuration  file=package.json ("build" field)
  • loaded parent configuration  preset=react-cra
  • writing effective config  file=dist/builder-effective-config.yaml
  • rebuilding native dependencies  dependencies=canvas@2.8.0, fsevents@1.2.13, node-expat@2.4.0 platform=darwin arch=x64
  • rebuilding native dependency  name=canvas version=2.8.0
  • rebuilding native dependency  name=fsevents version=1.2.13
  • rebuilding native dependency  name=node-expat version=2.4.0
  ⨯ cannot execute  cause=exit status 1
$ node-pre-gyp install --fallback-to-build
    Failed to execute 'node-gyp configure --fallback-to-build --module=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=electron-v16.0 --python=/usr/local/opt/python@3.9/bin/python3' (1)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    
                    errorOut=node-pre-gyp info it worked if it ends with ok
    node-pre-gyp info using node-pre-gyp@1.0.7
    node-pre-gyp info using node@17.0.1 | darwin | x64
    node-pre-gyp http GET https://github.com/Automattic/node-canvas/releases/download/v2.8.0/canvas-v2.8.0-electron-v16.0-darwin-unknown-x64.tar.gz
    node-pre-gyp ERR! install response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.8.0/canvas-v2.8.0-electron-v16.0-darwin-unknown-x64.tar.gz 
    node-pre-gyp WARN Pre-built binaries not installable for canvas@2.8.0 and electron@16.0.1 (electron-v16.0 ABI, unknown) (falling back to source compile with node-gyp) 
    node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.8.0/canvas-v2.8.0-electron-v16.0-darwin-unknown-x64.tar.gz 
    gyp info it worked if it ends with ok
    gyp info using node-gyp@8.2.0
    gyp info using node@17.0.1 | darwin | x64
    gyp info ok 
    gyp info it worked if it ends with ok
    gyp info using node-gyp@8.2.0
    gyp info using node@17.0.1 | darwin | x64
    gyp info find Python using Python version 3.9.8 found at "/usr/local/opt/python@3.9/bin/python3.9"
    gyp info spawn /usr/local/opt/python@3.9/bin/python3.9
    gyp info spawn args [
    gyp info spawn args   '/usr/local/Cellar/node/17.0.1/libexec/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
    gyp info spawn args   'binding.gyp',
    gyp info spawn args   '-f',
    gyp info spawn args   'make',
    gyp info spawn args   '-I',
    gyp info spawn args   '/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/config.gypi',
    gyp info spawn args   '-I',
    gyp info spawn args   '/usr/local/Cellar/node/17.0.1/libexec/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
    gyp info spawn args   '-I',
    gyp info spawn args   '/Users/xiaoxingcheng/.electron-gyp/16.0.1/include/node/common.gypi',
    gyp info spawn args   '-Dlibrary=shared_library',
    gyp info spawn args   '-Dvisibility=default',
    gyp info spawn args   '-Dnode_root_dir=/Users/xiaoxingcheng/.electron-gyp/16.0.1',
    gyp info spawn args   '-Dnode_gyp_dir=/usr/local/Cellar/node/17.0.1/libexec/lib/node_modules/npm/node_modules/node-gyp',
    gyp info spawn args   '-Dnode_lib_file=/Users/xiaoxingcheng/.electron-gyp/16.0.1/<(target_arch)/node.lib',
    gyp info spawn args   '-Dmodule_root_dir=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas',
    gyp info spawn args   '-Dnode_engine=v8',
    gyp info spawn args   '--depth=.',
    gyp info spawn args   '--no-parallel',
    gyp info spawn args   '--generator-output',
    gyp info spawn args   'build',
    gyp info spawn args   '-Goutput_dir=.'
    gyp info spawn args ]
    gyp: name 'openssl_fips' is not defined while evaluating condition 'openssl_fips != ""' in binding.gyp while trying to load binding.gyp
    gyp ERR! configure error 
    gyp ERR! stack Error: `gyp` failed with exit code: 1
    gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/Cellar/node/17.0.1/libexec/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:353:16)
    gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
    gyp ERR! System Darwin 21.1.0
    gyp ERR! command "/usr/local/Cellar/node/17.0.1/bin/node" "/usr/local/Cellar/node/17.0.1/libexec/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/Release/canvas.node" "--module_name=canvas" "--module_path=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/Release" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=electron-v16.0" "--python=/usr/local/opt/python@3.9/bin/python3"
    gyp ERR! cwd /Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas
    gyp ERR! node -v v17.0.1
    gyp ERR! node-gyp -v v8.2.0
    gyp ERR! not ok 
    node-pre-gyp ERR! build error 
    node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp configure --fallback-to-build --module=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=electron-v16.0 --python=/usr/local/opt/python@3.9/bin/python3' (1)
    node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
    node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
    node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1062:16)
    node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
    node-pre-gyp ERR! System Darwin 21.1.0
    node-pre-gyp ERR! command "/usr/local/Cellar/node/17.0.1/bin/node" "/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
    node-pre-gyp ERR! cwd /Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas
    node-pre-gyp ERR! node -v v17.0.1
    node-pre-gyp ERR! node-pre-gyp -v v1.0.7
    node-pre-gyp ERR! not ok 
error Command failed with exit code 1.
    
                    command=/usr/local/Cellar/node/17.0.1/bin/node /usr/local/Cellar/yarn/1.22.17/libexec/bin/yarn.js run install
                    workingDir=/Users/xiaoxingcheng/Code/suan-pc-grain-growth/gui/node_modules/canvas
error Command failed with exit code 1.
```
## Your Environment
* canvas@2.8.0
* node v17.0.1 on Mac OS X 12.0.1

**Comment by shuntatakemoto** [comment]
I'm in the same situation.

**Comment by basmilius** [comment]
Same here on Monterey, MacBook Pro M1 Max.

**Comment by freddneos** [comment]
Any update about that ?
facing the same M1 Pro , Monterey

**Comment by freddneos** [comment]
M1 native does not release binary version of canvas in node15. Therefore, when you install canvas, you will automatically compile from the source code, but if you do not install the corresponding dependency, you will report an error.

Solution

`arch -arm64 brew install pkg-config cairo pango jpeg giflib librsvg`

Worked for Me!!!

**Comment by landsman** [comment]
@freddneos thanks, working! 👏

**Comment by LinusU** [comment]
Native M1 building was fixed in 2.9.0, so hopefully the OP should be resolved 👍
